facebook-commerce.php

```diff
@@ -952,29 +952,34 @@ public function on_product_delete( $product_id ) {
 			return;
 		}
 
-		if ( ! $product->is_type( 'variable' ) ) {
+		if ( $product->is_type( 'variation' ) ) {
 
-			$fb_product_id = $this->get_product_fbid( self::FB_PRODUCT_ITEM_ID, $product_id );
+			$retailer_id = \WC_Facebookcommerce_Utils::get_fb_retailer_id( $product );
 
-			if ( $fb_product_id ) {
-				$this->delete_product_item( $product_id );
-			}
+			// enqueue variation to be deleted in the background
+			facebook_for_woocommerce()->get_products_sync_handler()->delete_products( [ $retailer_id ] );
 
-		} else {
+		} elseif ( $product->is_type( 'variable' ) ) {
 
-			$product_ids = array_merge( [ $product_id ], $product->get_children() );
+			$retailer_ids = [];
 
-			facebook_for_woocommerce()->get_products_sync_handler()->delete_products( $product_ids );
-		}
+			foreach ( $product->get_children() as $variation_id ) {
 
+				$variation = wc_get_product( $variation_id );
 
-		$fb_product_group_id = $this->get_product_fbid( self::FB_PRODUCT_GROUP_ID, $product_id );
+				if ( $variation instanceof \WC_Product ) {
+					$retailer_ids[] = \WC_Facebookcommerce_Utils::get_fb_retailer_id( $variation );
+				}
+			}
 
-		if ( $fb_product_group_id ) {
+			facebook_for_woocommerce()->get_products_sync_handler()->delete_products( $retailer_ids );
```

Comment:
```suggestion
                         // enqueue variations to be deleted in the background
			facebook_for_woocommerce()->get_products_sync_handler()->delete_products( $retailer_ids );
```

Reply:
Spacing may be off on my suggestion...

Reply:
The suggestion looks good. I tweaked the spacing and committed it.